File: tests/cases-py2many/with_cntx_manager.jl
using tempfile: NamedTemporaryFile
using textwrap: wrap
abstract type AbstractFileOp end
mutable struct FileOp <: AbstractFileOp
    file_name
    method
    file
end
function __enter__(self::FileOp)
    self.file = open(self.file_name, self.method)
    return self.file
end

function __exit__(self::FileOp, type_ = nothing, value = nothing, traceback = nothing)
    close(self.file)
end

if abspath(PROGRAM_FILE) == @__FILE__
    NamedTempFile::new() do temp_file
        file_path = temp_file.name
        FileOp(file_path, "w") do file
            write(file, "test")
        end
        open(file_path, "r") do f
            @assert(read(f) == "test")
        end
    end
end

File: tests/cases/with_cntx_manager.py
from tempfile import NamedTemporaryFile
from textwrap import wrap

# Testing context manager
class FileOp:
    def __init__(self, file_name, method):
        self.file_name = file_name
        self.method = method

    def __enter__(self):
        self.file = open(self.file_name, self.method)
        return self.file

    def __exit__(self, type=None, value=None, traceback=None):
        self.file.close()


if __name__ == "__main__":
    with NamedTemporaryFile(mode="a+", delete=False) as temp_file:
        file_path = temp_file.name
        with FileOp(file_path, "w") as file:
            file.write("test")
        with open(file_path, "r") as f:
            assert f.read() == "test"
